### PR TITLE
PG-1236 Community Email Subscription Settings

### PR DESCRIPTION
**Pull request by wsbastian**
**Reminder to create the PR against this repository, and not the base fork of `rafalp/Misago`.**

### JIRA Ticket

https://bighealth.atlassian.net/browse/PG-1236

### Problem

As a user, I would like the emails I receive from community notifying me of updates to threads I’m subscribed to to contain a link that allows me to easily manage my Community email subscription settings, so I can opt-in and out of email notifications quickly and easily

### Solution

* We have added a new component to the /subscribed page in Community to allow bulk sub, sub-email, and unsub
* We have updated the subscription email templates to include a link to the subscription page
* We have updated the login redirect URL generation to follow the url formatting to allow redirection back to Community

#### Unit tests + Integration Tests

* updated test_auth_middleware.py to validate new redirect point

#### Manual Tests

* We won't actually see redirection back to /subscriptions when a user logs in after clicking the link in their email, we should instead validate that the redirect URL is structured correctly. See https://bighealth.atlassian.net/browse/PG-1501 and https://bighealth.atlassian.net/browse/ENR-31

### CODE REVIEW

File: frontend/src/services/store.js [hunk 1]
```diff
@@ -15,7 +15,9 @@ export class StoreWrapper {
   init() {
     this._store = createStore(
       combineReducers(this._reducers),
-      this._initialState
+      this._initialState,
+      // TODO: Remove this when live
+      window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
```
Comment (avi-bh) [inline frontend/src/services/store.js:20]: what is this for?

**Review by avi-bh — DISMISSED** [review]
Nice! I looked over the react code, but I'll let @jjosephbh review them since she'll know what she's looking at

File: frontend/src/services/store.js (hunk 1 above)
Comment (wsbastian) [reply frontend/src/services/store.js:20]: It lets us hook redux devtools into our application state so we can visualize state structure.

File: frontend/src/components/threads/subscription.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  return (
+    <ul className={props.dropdownClassName || "dropdown-menu stick-to-bottom"}>
+      <Disable {...props} />
+      <Enable {...props} />
+      <Email {...props} />
+    </ul>
+  )
+}
+
+export class Disable extends React.Component {
```
Comment (jjosephbh) [inline frontend/src/components/threads/subscription.js:37]: Just a nit: Is this a button? The name does not really tell me what the component is really.

File: frontend/src/components/threads/subscription.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        <button className="btn btn-link" onClick={this.onClick}>
+          <span className="material-icon">star</span>
+          {gettext("Subscribe E-mail All")}
+        </button>
+      </li>
+    )
+  }
+}
+
+export function update(props, newState, value) {
```
Comment (jjosephbh) [inline frontend/src/components/threads/subscription.js:100]: Trying to understand what this function is for. I am guessing you are updating the state with the changes? If so why don't you `setState` and then use tie that to the dispatch?

**Review by jjosephbh — CHANGES_REQUESTED** [review]

File: frontend/src/components/threads/subscription.js (hunk 2 above)
Comment (wsbastian) [reply frontend/src/components/threads/subscription.js:37]: Yeah, this was largely provisioned from the in-thread subscription example at https://github.com/sleepio/community-app-web/blob/f3d8b5a88f68621a284f47f6bdfcc65a6b6092bf/frontend/src/components/thread/subscription.js#L59-L59, and the naming came with.

File: frontend/src/components/threads/subscription.js (hunk 3 above)
Comment (wsbastian) [reply frontend/src/components/threads/subscription.js:100]: Also following the update model used by Misago from the and single-thread view at: 
https://github.com/sleepio/community-app-web/blob/f3d8b5a88f68621a284f47f6bdfcc65a6b6092bf/frontend/src/components/thread/subscription.js#L59-L59

Our change here was to modify what we're updating, and how we're updating them.

Being new to react, I don't entirely follow what the suggested approach would entail.

Comment (jjosephbh) [reply frontend/src/components/threads/subscription.js:100]: Hmm ok so you are following their code. OK I will let that go.

File: frontend/src/components/threads/subscription.js (hunk 2 above)
Comment (jjosephbh) [reply frontend/src/components/threads/subscription.js:37]: Sigh I swear. Ok.

File: frontend/src/components/threads/subscription.js (hunk 3 above)
Comment (wsbastian) [reply frontend/src/components/threads/subscription.js:100]: I'm interested to learn more about what you're preferred approach would be. If not within this PR than in a sidebar convo, if you've got the time.

File: frontend/src/components/threads/subscription.js [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        {
+          op: "replace",
+          path: "subscription",
+          value: value
+        }
+      ]
+    })
+
+    .then(
+      // this feels redundant, but it's also what's being done in single thread.
```
Comment (jjosephbh) [inline frontend/src/components/threads/subscription.js:124]: It feels because it is lol

**Review by jjosephbh — DISMISSED** [review]

**Review by jjosephbh — APPROVED** [review]

**Pull request merged**